t/lib/Test/Act/Environment.pm

```diff
@@ -0,0 +1,221 @@
+# ABSTRACT: Create an environment for testing the Act application
+use 5.20.0;
+package Test::Act::Environment;
+
+use Moo;
+use Types::Standard qw(InstanceOf Str);
+use namespace::clean;
+
+use feature qw(signatures);
+no warnings qw(experimental::signatures);
+
+use File::Temp qw(tempdir);
+use File::Copy::Recursive qw(dircopy);
+
+use FindBin qw($RealBin);
+FindBin::again();
+
+use Test::WWW::Mechanize::PSGI;
+
+use Test::Lib;
+use Test::Act::SMTP::Server;
+
+my $tempdir;
+
+# This needs to happen _before_ Act::Config is used...
+BEGIN {
+    $tempdir = File::Temp->newdir( CLEANUP => 0 );
+    # The following will need adaption to the post-november master
+    # Part 1: Link the distribution files from the repository so that
+    # changes in the working tree become immediately effective
+    for my $dir (qw(templates po wwwdocs)) {
+        symlink "$RealBin/../$dir","$tempdir/$dir"
+            or die "Failed to create a symlink for '$RealBin/../$dir': '$!'";
+    }
+    # Part 2: Copy the files from the test environment
+    for my $dir (qw(actdocs conf conferences)) {
+        dircopy "$RealBin/acthome/$dir","$tempdir/$dir";
+    }
+    # Part 3: These must just exist
+    for my $dir (qw(photos ttc)) {
+        mkdir "$tempdir/$dir" or die "Could not create '$tempdir/$dir': '$!'";
+    }
+    $ENV{ACTHOME} = "$tempdir";
+}
+
+use Act::Config;
+
+my $smtp_server = Test::Act::SMTP::Server->instance;
+$Config->set(email_hostname => 'localhost');
+$Config->set(email_port     => $smtp_server->port);
+
+
+has mech => (
+    is => 'rwp', isa => InstanceOf['Test::WWW::Mechanize'],
+    builder => 'build_mech',
+    documentation =>
+        'A test client for the application tests',
+);
+
+sub build_mech ($self) {
+    require Act::Dispatcher;
+    my $mech = Test::WWW::Mechanize::PSGI->new(app => Act::Dispatcher->to_app);
+    $self->_set_mech($mech);
+    return $mech;
+}
+
+has base => (
+    is => 'ro', isa => Str,
+    builder => '_build_base',
+    documentation =>
+        'The base URL for tests',
+);
+
+sub _build_base ($self) {
+    my $host = `hostname`;  chomp $host;
+    my $port = $ENV{ACT_TEST_PORT} || 5050;
```

Comment:
in the act.ini this port is 5000

Reply:
Ah, I'll changed this in the conference's act.ini.  It isn't relevant right now since the tests are run without a network.
The port is different from 5000 because I usually have always an Act instance listening on port 5000 and still want to run automated tests.